PythonVisualizations/SortingBase.py

```diff
@@ -0,0 +1,420 @@
+import random
+import time
+from tkinter import *
+try:
+    from drawable import *
+    from VisualizationApp import *
+except ModuleNotFoundError:
+    from .drawable import *
+    from .VisualizationApp import *
+ 
+
+class SortingBase(VisualizationApp):
+    CELL_SIZE = 50
+    CELL_BORDER = 2
+    CELL_BORDER_COLOR = 'black'
+    ARRAY_X0 = 100
+    ARRAY_Y0 = 100
+    FOUND_COLOR = 'brown4'
+    nextColor = 0
+    CELL_WIDTH = CELL_SIZE
+    changeSize = False 
+    
+    def __init__(self, size=10, title="SortingBase", **kwargs):
+        super().__init__(title=title, **kwargs)
+        self.size = size
+        self.title = title
+        self.list = []  # Internal array of drawable cell values
+        
+        #self.buttons = self.makeButtons()
+        for i in range(size):
+            self.list.append(drawable(random.randrange(99)))
+        self.display()
```

Comment:
The constructor for the `SortingBase` class should rely more on its subclasses to configure its behavior.  

* It should not provide a default value for the `title` parameter.  That should be left to the subclass to provide.
* It should not fill the initial array with random values.  It should start off with an empty array of the given `size`.  The subclasses should decide what initial values are inserted, if any.

Reply:
Implemented in commit dde9c81adda38c9692a76c9773cd8c2e2baa32b8

Reply:
Title fix was in commit 43b212d9abdfbebd3438d8e724f89e5adde80f6f
Initialization of array in AdvancedSorting in commit 9aef4022364fdc31e00f5246ac8dd2668f7bbe31
Removal of initialization of array in SortingBase in commit befa7681d81c81ec3d2fe0e23cbe80f914f1db0a

Reply:
I made a correction to the changes to the `title` parameter in commit 2e0ff28.